src/main/kotlin/ca/kittle/capabilities/party/CreateParty.kt

```diff
@@ -16,28 +18,34 @@ import mu.KotlinLogging
 
 private val logger = KotlinLogging.logger {}
 
-context(MongoDatabase)
-fun Routing.createParty() {
+/**
+ * Route to create a new Party
+ */
+context(MongoDatabase) fun Routing.createParty() {
     post("/party") {
-        val party: Party = call.receive()
-        val id = createParty(party).getOrThrow()
-        call.respondRedirect("/party/$id")
+        try {
+            val party: Party = call.receive()
+            val id = createParty(party).getOrThrow()
+            call.respondRedirect("/party/$id")
+        } catch (e: Exception) {
+            call.respond(HttpStatusCode.BadRequest, StatusErrorMessage("Failed to create party"))
+        }
     }
 }
 
-context(MongoDatabase)
-suspend fun createParty(party: Party): Result<String> =
+context(MongoDatabase) suspend fun createParty(party: Party): Result<String> =
     withContext(Dispatchers.IO) {
         runCatching {
             val collection = this@MongoDatabase.getCollection(PartyEntity.COLLECTION_NAME)
-            collection.find(Filters.eq(Party::name.name, party.name)).first()
-                ?.let(PartyEntity::fromDocument)?.id
-                ?: run {
-                    collection.insertOne(party.toDocument())
-                    party.id
-                }
+            val existingParty = collection.find(Filters.eq(Party::name.name, party.name.value)).first()
```

Comment:
**issue (bug_risk):** Be cautious of potential race conditions when checking for existing parties

The current implementation checks for an existing party before creating a new one. This could lead to race conditions if multiple requests try to create parties with the same name simultaneously. Consider using a unique index on the party name and handling duplicate key errors instead.